docs/DeveloperGuide.md

```diff
@@ -170,6 +170,34 @@ sequence diagram shows the interactions within `Model` when editing a person.
 
 ![EditSequenceDiagram-Model](images/EditSequenceDiagram-Model.png)
 
+### Remark Feature
+
+Users are able to add a `Remark` to a `Person` in FINDvisor to note down some information about a `Person`.
+
+The remark feature is implemented through creating a `RemarkCommand`, which updates the `Remark` of a `Person`.
+A separate command is used to support the remark feature to separate the logic of the personal particulars of a `Person` from the `Remark`.
+
+`Remark` is implemented with the use of the `Optional` generic class (i.e. `Optional<Remark>`) as it is an optional attribute of a `Person`.
+While it is possible to determine an empty `Remark` through its value, the `Optional` generic class provides a better abstraction for when a `Remark` is empty.
+
+When a user passes a parameter that is either empty or consists exclusively of whitespace, the `Remark` attribute of a `Person` would be updated to `Optional.empty()`.
+This is equivalent to a user removing the previous `Remark` of a `Person`.
+
+The following sequence diagram shows how the remark value is parsed through the `Logic` component:
+
+![RemarkSequenceDiagram-Logic](images/RemarkSequenceDiagram-Logic.svg)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `RemarkCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
```

Comment:
Also mentioned in @Dethada PR [here](https://github.com/AY2324S2-CS2103-F15-1/tp/pull/135#discussion_r1539171876), since this is applicable for all sequence diagrams perhaps would be better to move this up when we are finalising our DG